pkgs/development/tools/rust/rust-analyzer/generic.nix

```diff
@@ -17,26 +18,21 @@ rustPlatform.buildRustPackage {
 
   buildAndTestSubdir = "crates/rust-analyzer";
 
-  cargoBuildFlags = lib.optional useJemalloc "--features=jemalloc";
+  cargoBuildFlags = lib.optional useMimalloc "--features=mimalloc";
 
-  nativeBuildInputs = lib.optionals doCheck [ rustPlatform.rustcSrc ];
+  nativeBuildInputs = lib.optional useMimalloc cmake;
```

Comment:
Note: Needed for `mimalloc_rust` to build its bundled version of the mimalloc library. https://github.com/purpleprotocol/mimalloc_rust#requirements